text_scrubber/geo/geo.py

```diff
@@ -0,0 +1,336 @@
+
+    # There's a number of known expansions/translations which can be applied. Check if we can find anything with that
+    if cleaned_country in _COUNTRY_RESOURCES['replacements']:
+        candidate = _COUNTRY_RESOURCES['cleaned_to_capitilized'][_COUNTRY_RESOURCES['replacements'][cleaned_country]]
+        return [(candidate, 1.0) if return_scores else candidate]
+
+    # Check if the country follows a certain country pattern
+    known_country = pattern_match(country, _COUNTRY_RESOURCES['replacement_patterns'])
+    if known_country:
+        candidate = _COUNTRY_RESOURCES['cleaned_to_capitilized'][known_country]
+        return [(candidate, 1.0) if return_scores else candidate]
+
+    # Check if we can find a close match (using default threshold of 0.8 (magic number))
+    country_match = find_closest_string(cleaned_country, _COUNTRY_RESOURCES['cleaned_trigrams'])
+    if country_match:
+        best_matches, score = country_match
+        candidates = (_COUNTRY_RESOURCES['cleaned_to_capitilized'][country] for country in best_matches)
+        return [(candidate, score) for candidate in candidates] if return_scores else list(candidates)
+
+    # No match found
+    return []
+
+
+def normalize_state(state: str, return_scores: bool = False) -> List[Union[Tuple[str, str, float], Tuple[str, str]]]:
+    """
+    Cleans up a state by string cleaning and performs some basic state lookups to get the canonical name.
+
+    :param state: State name or code.
+    :param return_scores: Whether to return match scores.
+    :return: List of (state, country) candidates in canonical form.
+    """
+    # Clean state
+    cleaned_state = clean_state(state)
+
+    # Check if state is part of the known states list
+    if cleaned_state in _STATE_RESOURCES['state_country_map']:
+        candidates = _STATE_RESOURCES['state_country_map'][cleaned_state]
+        candidates = [(candidate, 1.0) for candidate in candidates] if return_scores else candidates
+
+    # Check if we can find a close match (using default threshold of 0.8 (magic number))
+    else:
+        state_match = find_closest_string(cleaned_state, _STATE_RESOURCES['cleaned_trigrams'])
+        if state_match:
+            best_matches, score = state_match
+            candidates = (candidate for state in best_matches
+                          for candidate in _STATE_RESOURCES['state_country_map'][state])
+            candidates = [(candidate, score) for candidate in candidates] if return_scores else list(candidates)
+        else:
+            # No match found
+            candidates = []
+
+    # Return canonical form
+    if return_scores:
+        return sorted((state, _COUNTRY_RESOURCES['cleaned_to_capitilized'][country], score)
+                      for (state, country), score in candidates)
+    else:
+        return sorted((state, _COUNTRY_RESOURCES['cleaned_to_capitilized'][country]) for state, country in candidates)
+
+
+def normalize_city(
```

Comment:
I expected a `restrict_countries` parameter here. I.e., you should be able to call `normalize_city('Amsterdam', restrict_countries=['NL'])` and get only the result `('Amsterdam', 'Netherlands', 1.0)`. 

You should be able to either provide a country string or the country code (`'NL'`/`'Netherlands'`) to `restrict_countries`